### PR TITLE
Fix random state

### PR DESCRIPTION
**Pull request by jasmainak**
There is still the issue with `mne.compute_covariance` though ...

### CODE REVIEW

File: scripts/processing/Makefile [hunk 1]
```diff
@@ -2,15 +2,20 @@ PYTHON = python
...
+	@echo "  check         check if the dependencies are available"
 	@echo "  fetch-recon   to fetch the data and run anatomy scripts"
-	@echo "  preproc        to run all scripts after anatomy. LFREQ must be set for highpass cutoff"
+	@echo "  preproc       to run all scripts after anatomy. LFREQ must be set for highpass cutoff"
 	@echo "  preproc-log   to send the output through tee also to a logfile"
-	@echo "  profile        to profile memory consumption"
-	@echo "  all        to build fetch data, run anatomy scripts and all the processing scripts with no highpass"
+	@echo "  profile       to profile memory consumption"
+	@echo "  all           to build fetch data, run anatomy scripts and all the processing scripts with no highpass"
+	@echo "  clean		   remove intermediate scripts generated by processing"
```
Comment (agramfort) [inline scripts/processing/Makefile]: too many spaces before remove

**Comment by jasmainak** [comment]
okay I think `random_state` for cov estimation is also fixed now, anything else I might have missed?

**Comment by larsoner** [comment]
https://github.com/mne-tools/mne-biomag-group-demo/blob/master/scripts/results/statistics/plot_sensor_spatio_temporal_cluster_stats.py#L67

**Comment by jasmainak** [comment]
Thanks! Good catch

File: scripts/processing/08-make_cov.py [hunk 2]
```diff
@@ -30,7 +31,8 @@ def run_covariance(subject_id, tsss=False):
                             % (subject, l_freq))
     print('  Computing regularized covariance')
     epochs = mne.read_epochs(fname_epo, preload=True)
-    cov = mne.compute_covariance(epochs, tmax=0, method='shrunk')
+    cv = ShuffleSplit(3, random_state=random_state)
```
Comment (agramfort) [inline scripts/processing/08-make_cov.py]: @dengemann I think we should use k-fold as there is a lot of autocorrelation in data.

**Comment by dengemann** [comment]
I would use the default here. Which is k-fold. Why change defaults?

**Comment by agramfort** [comment]
I agree kfold is deterministic (shuffle=False by default) so let's not
change this.

**Comment by jasmainak** [comment]
The docstring in MNE says: `Defaults to 3, which will internally trigger a default 3-fold shuffle split.`

File: scripts/processing/08-make_cov.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def run_covariance(subject_id, tsss=False):
                             % (subject, l_freq))
     print('  Computing regularized covariance')
     epochs = mne.read_epochs(fname_epo, preload=True)
-    cv = ShuffleSplit(3, random_state=random_state)
+    cv = KFold(3, random_state=random_state)
```
Comment (larsoner) [inline scripts/processing/08-make_cov.py]: I think the docs in MNE are wrong:

http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html

Can you confirm and open a PR to fix them?

**Comment by dengemann** [comment]
Yes, it's true. It should be kfold in the docs. Actually the default of
cross_val_score in sklearn.

File: scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py [hunk 4]
```diff
@@ -72,7 +72,7 @@
 cluster_stats = permutation_cluster_1samp_test(
     data, threshold=threshold, n_jobs=N_JOBS, verbose=True, tail=tail,
     step_down_p=0.05, connectivity=connectivity,
-    n_permutations=n_permutations)
+    n_permutations=n_permutations, seed=random_state)
```
Comment (agramfort) [inline scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py:75]: if our results are actually sensititve to this seed then we need to do more permutations.

File: scripts/processing/08-make_cov.py [hunk 5]
```diff
@@ -30,7 +31,8 @@ def run_covariance(subject_id, tsss=False):
                             % (subject, l_freq))
     print('  Computing regularized covariance')
     epochs = mne.read_epochs(fname_epo, preload=True)
-    cov = mne.compute_covariance(epochs, tmax=0, method='shrunk')
+    cv = KFold(3, random_state=random_state)
```
Comment (agramfort) [inline scripts/processing/08-make_cov.py]: is it really necessary. For me it's deterministic by default. can you confirm?

Comment (jasmainak) [reply scripts/processing/08-make_cov.py]: yes I think it's deterministic by default but I added it just to be explicit

Comment (larsoner) [reply scripts/processing/08-make_cov.py]: I'd rather add a comment saying this uses CV but it's deterministic, rather than add code (a comment would be needed even with the code to explain why, otherwise people might assume it is necessary to get KFold CV)

File: scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py (hunk 4 above)
Comment (larsoner) [reply scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py:75]: I don't expect them to be very sensitive, but it's good to fix it to move toward 100% reproducibility

File: scripts/processing/08-make_cov.py (hunk 5 above)
Comment (agramfort) [reply scripts/processing/08-make_cov.py]: could it explain the differneces on results between yours and mainak?

Comment (agramfort) [reply scripts/processing/08-make_cov.py]: +1

File: scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py (hunk 4 above)
Comment (agramfort) [reply scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py:75]: could it explain the differnece of results between yours and mainak?

Comment (larsoner) [reply scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py:75]: I doubt it, I think I still get different results even with `seed` set (but need to double-check)

Comment (jasmainak) [reply scripts/results/statistics/plot_sensor_cluster_stats_eeg_channel.py:75]: not really because this wasn't the file which was showing the difference. In the other one, the `random_seed` was already set.

File: scripts/processing/08-make_cov.py (hunk 5 above)
Comment (jasmainak) [reply scripts/processing/08-make_cov.py]: umm ... I actually prefer to leave it like this because we don't have tests. I can add a comment to explain why it's like this though. When I see `cv=3` it doesn't tell me if it's deterministic or not. I have to either look at the code or go to scikit-learn documentation ...

**Comment by jasmainak** [comment]
should we merge this so that everyone is working on the same version of code at least? @agramfort are you running on this branch or on `master`?

**Comment by agramfort** [comment]
for now I ran master but I still cannot build the doc as I get a segfault :(

I look into it when I find 5 mins ...

**Comment by agramfort** [comment]
let me merge and see
